File: test/fixtures/each.js
(function(){
  _.each = function(obj, iterator, context) {
    var index = 0;
    try {
      if (obj.forEach) {
        return obj.forEach(iterator, context);
      }
      if (_.isArray(obj) || _.isArguments(obj)) {
        var a = obj;
        var d = [];
        for (var b=0, c=a.length; b<c; b++) {
          var item = a[b];
          var i = b;
          d[b] = iterator.call(context, item, i, obj);
        }
        return d;
      }
      var e = _.keys(obj);
      for (var f=0, g=e.length; f<g; f++) {
        var key = e[f];
        iterator.call(context, obj[key], key, obj);
      }
    } catch (e) {
      if (e !== breaker) {
        throw e;
      }
    }
    return obj;
  };
})();

File: test/fixtures/each.cs
# The cornerstone, an each implementation.
# Handles objects implementing forEach, arrays, and raw objects.
_.each: obj, iterator, context =>
  index: 0
  try
    return obj.forEach(iterator, context) if obj.forEach
    return iterator.call(context, item, i, obj) for item, i in obj. if _.isArray(obj) or _.isArguments(obj)
    iterator.call(context, obj[key], key, obj) for key in _.keys(obj).
  catch e
    throw e if e aint breaker.
  obj.